src/sdk/namespace_fs.js

```diff
@@ -3017,8 +3033,12 @@ class NamespaceFS {
         const is_gpfs = native_fs_utils._is_gpfs(fs_context);
         let retries = config.NSFS_RENAME_RETRIES;
         let latest_ver_info;
+        const is_key_dir_path = await this._is_key_dir_path(fs_context, params.key);
         for (;;) {
             try {
+                if (is_key_dir_path && !this._is_directory_content(latest_ver_path, params.key)) {
```

Comment:
@naveenpaul1 @shirady Have you checked also what happens when trying to delete a/b/c/?

Reply:
we probably want to avoid versioning flow on this case as well, right?

Reply:
I don't remember that I checked this case - delete a/b/c without the beginning slash instead of /ab/c/ (I thought only about deleting/a/b/c without the ending slash instead of /a/b/c/ as mentioned in the issue).

Reply:
@romayalon @shirady for`a/b/c/`, with this code we are just deleting the `.folder` file not creating any delet_marker,

Reply:
@naveenpaul1 can you point me to the code that does that?

Reply:
@romayalon here : [link](https://github.com/noobaa/noobaa-core/pull/8480/files#diff-74a1674f40effb24da2459330c005cd5ab9fda4e1bca500a4387607926a2af1fR3058) we are skipping copy file to new version when customer try to delete the dir. and else block get executed which do simple deletion and e are also skipping delete_marker creation if is_key_dir_path true.

Reply:
@romayalon we are deleting the `.folder` file when user pass key `a/b/c/` we are not creating the version and delete_marker